src/_nebari/config_set.py

```diff
@@ -0,0 +1,51 @@
+import pathlib
+
+from packaging.requirements import Requirement
+from pydantic import BaseModel, ConfigDict, field_validator
+
+from _nebari._version import __version__
+from _nebari.utils import yaml
+
+
+class ConfigSetMetadata(BaseModel):
+    model_config: ConfigDict = ConfigDict(extra="allow", arbitrary_types_allowed=True)
+    name: str  # for use with guided init
```

Comment:
If `none` is allowed here, shouldn't the type be `Optional[str] = None`?

Reply:
agreed